python_modules/libraries/dagster-dbt/dagster_dbt/cloud_v2/run_handler.py

```diff
@@ -195,7 +196,25 @@ def to_default_asset_events(
                     test_unique_id=unique_id,
                 )
 
-                if asset_check_key is not None:
+                if (
+                    context
+                    and has_asset_def
+                    and asset_check_key is not None
+                    and asset_check_key in context.selected_asset_check_keys
+                ):
+                    # The test is an asset check in an asset, so yield an `AssetCheckResult`.
+                    yield AssetCheckResult(
+                        passed=result_status == TestStatus.Pass,
+                        asset_key=asset_check_key.asset_key,
+                        check_name=asset_check_key.name,
+                        metadata=metadata,
+                        severity=(
+                            AssetCheckSeverity.WARN
+                            if result_status == TestStatus.Warn
+                            else AssetCheckSeverity.ERROR
+                        ),
+                    )
+                elif not has_asset_def and asset_check_key is not None:
```

Comment:
I still really don't understand this line of code. Like; why is it ok to yield an asset check evaluation in a non asset context? Is it because we don't coerce them?

Reply:
This behavior was introduced [here](https://github.com/dagster-io/dagster/pull/28481). We do the same for dbt core, the goal is to yield an asset check evaluation when the asset_check_key is not none.

I agree though that we should consolidate to ease maintenance.